### PR TITLE
chore: only use @Nonnull and ban similar annotations

### PR DESCRIPTION
**Pull request by jbee**
Uses `@Nonnull` (javax annotation) instead of `@NotNull` (javax validation) or `@NonNull` (lombok).

Switching to `@Nonnull` also caused hibernate to run nullness analysis in tests so that all non-null database fields need to have a value when stored. This required some more changes to tests to provide a value.

In case of `IndicatorGroup.groupSet` this mapping was not updated similar to what had been done to `DateElementGroup`.
A `IndicatorGroup` can be member of multiple `IndicatorGroupSet`s. Therefore the `IndicatorGroup.groupSet` was changed to a `Set<IndicatorGroupSet>` called `IndicatorGroup.groupSets`. This change was mande now as hibernate otherwise considered the relation as non-null and complained about missing value in tests.

### CODE REVIEW

**Review by teleivo — APPROVED** [review]

**Review by enricocolasante — APPROVED** [review]

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/eventvisualization/SimpleDimensionHandler.java [hunk 1]
```diff
@@ -139,9 +139,7 @@ private List<BaseDimensionalItemObject> loadDimensionalItems( final String dimen
     private SimpleDimension createSimpleEventDimensionFor( final DimensionalObject dimensionalObject,
         final Attribute attribute )
     {
-        final SimpleDimension simpleDimension = new SimpleDimension();
```
Comment (teleivo) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/eventvisualization/SimpleDimensionHandler.java:142]: thank you very much 🎉

**Review by teleivo — APPROVED** [review]

**Review by teleivo — APPROVED** [review]

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/eventvisualization/SimpleDimensionHandler.java (hunk 1 above)
Comment (teleivo) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/eventvisualization/SimpleDimensionHandler.java:142]: annoying that simple things often lead to more problems 😬 constructors seem to have gone out of fashion at some point 😂

**Review by gnespolino — APPROVED** [review]
